galaxyproject/galaxy
# Beta History: homogeneous collection indication

This is part of the **Necessary fixes at the Collection level** for the History roadmap.

>The collapsed view should indicate whether this is a “homogeneous” (all datasets are of the same type) or a “heterogeneous” (datasets are of different types) collection.
A homogeneous collection should indicate datatype in the collapsed view.

## Backend changes
This PR adds a new field `elements_datatypes` to the DatasetCollections API response (currently only when using the *betawebclient* view) that contains a set with all the different datatypes of the collection elements.

## Frontend changes
The collections displayed in the Beta History panel now show the datatype in a small textual description when the collection is homogeneous. This information is encapsulated in its own component to be reused.


## How to test the changes?
  - Switch to the Beta History panel
  - Create various collections some homogeneous (containing the same kind of datasets) and some heterogeneous.
  - You should see the datatype of the datasets in the collapsed view when the history is homogeneous.
  
    ![Screenshot from 2021-12-16 19-09-30](https://user-images.githubusercontent.com/46503462/146426341-d16d07d8-164b-40ef-976a-52e73d94bfc9.png)



## License
